app/components/autocomplete/channel_mention/channel_mention.js

```diff
@@ -70,7 +73,49 @@ export default class ChannelMention extends PureComponent {
         if (matchTerm !== this.props.matchTerm) {
             // if the term changed and we haven't made the request do that first
             const {currentTeamId} = this.props;
-            this.props.actions.searchChannels(currentTeamId, matchTerm);
+            if (!isMinimumServerVersion(this.props.serverVersion, 5, 4)) {
+                this.props.actions.searchChannels(currentTeamId, matchTerm);
+                return;
+            }
+            this.props.actions.autocompleteChannelsForSearch(currentTeamId, matchTerm).then((response) => {
```

Comment:
should we throttle this somehow?

Reply:
We could debounce it, I was using the same pattern used for searchChannels call, but makes sense to debounce both.

Reply:
Added a debounce on run search.